modulus/distributed/manager.py

```diff
@@ -427,20 +430,20 @@ def create_process_subgroup(
     @staticmethod
     def create_orthogonal_process_group(
```

Comment:
What happens if I try to create two orthogonal process groups for a single `group_name`? From the code below, it appears that this behavior is not supported but does it raise an error? 

Related: After I create an orthogonal process group, how can I see that that group is orthogonal to the original process group? It just looks like this code creates another group with the right ranks but there isn't really another way to tell the relationship a posteriori.

Reply:
I think the latter will be quite hard to achieve. The `DistributedManager` here is typically only a lightweight wrapper around PyTorch's `ProcessGroup` which doesn't imply any constraints on using these things.

This PR mainly introduces the idea of a config to define the relation of process groups which IMHO should address the point you are raising here to some extent, right?

Reply:
You can create two orthogonal process groups for a single `group_name`. There's nothing wrong in that. But those two orthogonal groups would end up being equivalent.

For the second point, like @stadlmax mentions, it would be a bit tedious to verify orthogonality of two process groups. The way that you'd do it is to do a gather over the main group of all the ranks of the processes in the orthogonal process group and ensure that they are all the same. Does this make sense? We can add such a utility if there is a need for it.

Reply:
Currently, it will just create a new process group with the same ranks since the `DistributedManager` doesn't store a mapping of group to orthogonal_group. I'd suggest we add this in a separate PR since this is an isolated optimization that doesn't change functionality.